cds_migrator_kit/rdm/records/load/load.py

```diff
@@ -162,12 +163,23 @@ def _after_publish_update_dois(self, identity, record, entry):
                 _draft = current_rdm_records_service.edit(identity, record["id"])
                 current_rdm_records_service.publish(identity, _draft["id"])
 
-    def _after_publish_update_created(self, record, entry):
-        """Update created timestamp post publish."""
-        # Fix the `created` timestamp forcing the one from the legacy system
-        # Force the created date. This can be done after publish as the service
-        # overrides the `created` date otherwise.
-        record._record.model.created = arrow.get(entry["record"]["created"]).datetime
+    def _after_publish_update_created(self, record, entry, version):
+        """Update created timestamp post publish.
+
+        Ensures that the `created` timestamp is correctly set, preferring:
+        1. The original legacy system value for the version.
+        2. The record's creation date if there are no files.
+        3. Today's date if the original value and file creation date is missing.
+        """
+        creation_date = entry["record"]["created"].datetime
+
+        versions = entry.get("versions", {})
+        version_data = versions.get(version, {})
+        # Use records creation date for the first version
+        if version_data.get("files") and not version == 1:
```

Comment:
I think there is more cases than this, especially in the thesis:
it can happen that there was no creation date on the record (especially when created by inspire sync) even if it is a first version. could you take this into account and create a test case for it?

Reply:
That's already covered in the the transform logic, where it assigns now as the date if the record is missing it ([here reference](https://github.com/CERNDocumentServer/cds-migrator-kit/blob/master/cds_migrator_kit/rdm/records/transform/transform.py#L89)). I don't think we have a test for this, will add it 👍

Reply:
could you add a comment then explaining why subsequent versions use publication_date? it is slightly confusing, I know the publication_date is computed initially from the creation date of the file but here when we use it in this way, it is not intuitive. Maybe a comment would help.
I see we use datetime, instead of full timestamp. Wouldn't it be more accurate to register the full datetime=? is it possible?